tests/unit/test_store/fake_storage.py

```diff
@@ -1,107 +1,244 @@
+        for tmp_object in delete_object_list:
+            self.storage[bucket_name].pop(tmp_object._name, None)
+        return list()
+
+    def make_bucket(self, bucket_name, location=None, object_lock=False):
+        self.storage[bucket_name] = dict()
+
+    def get_object(self, bucket_name, object_name, offset=0, length=0, request_headers=None, ssec=None, version_id=None,
+                   extra_query_params=None):
+        if bucket_name in self.storage.keys():
+            if object_name in self.storage[bucket_name].keys():
+                return FakeHTTPMinioResponse(self.storage[bucket_name][object_name].content)
+        return FakeHTTPMinioResponse()
+
+    def bucket_exists(self, bucket_name):
+        return bucket_name in self.storage.keys()
+
+    def list_objects(self, bucket_name, prefix=None, recursive=False, start_after=None, include_user_meta=False,
+                     include_version=False, use_api_v1=False):
+        result = list()
+        if prefix is not None:
+            for key in self.storage[bucket_name].keys():
+                str_key = str(key)
+                if str_key.startswith(prefix):
+                    result.append(self.storage[bucket_name][key])
+        else:
+            result = list(self.storage[bucket_name].values())
+        return result
+
+
+class FakeObjectStore(ObjectStoreABC):
+
+    def __init__(self):
+        self._objects = dict()
+
+    def clear_storage(self, object_name_prefix=None):
+        tmp_dict = dict.fromkeys(self._objects.keys(), [])
+        for key in tmp_dict.keys():
+            str_key = str(key)
+            if str_key.startswith(object_name_prefix):
+                del self._objects[key]
+
+    def put_object(self, object_name: str, content) -> int:
+        self._objects[object_name] = content
+
+    def get_object(self, object_name: str):
+        if object_name in self._objects.keys():
+            return self._objects[object_name]
+        else:
+            return None
+
+    def list_objects(self, object_name_prefix: str):
+        list_result = []
+        for key in self._objects.keys():
+            str_key = str(key)
+            if str_key.startswith(object_name_prefix):
+                list_result.append(self._objects[key])
+        return list_result
+
+
+class FakeSecretsStore(SecretsStoreABC):
+    def get_secrets(self, path: str) -> dict:
+        secrets = {
+            "secret_path1":
+                {
+                    "secret1": "white",
+                    "secret2": "black"
+                },
+            "secret_path2":
+                {
+                    "secret3": "orange",
+                    "secret4": "green"
+                }
+
+        }
+        return secrets[path]
+
+
+class FakeElasticSearchClient(Elasticsearch):
+
+    def search(self, *, body: Optional[Any] = ..., index: Optional[Any] = ..., doc_type: Optional[Any] = ...,

```

Comment:
What is this ?! :))))))))))
This fake client is certainly way more versatile than what a fake shall do. Well done, anyways, for putting efforts into it :)